test/distributed/tensor/parallel/test_ddp_2d_parallel.py

```diff
@@ -0,0 +1,102 @@
+# Owner(s): ["oncall: distributed"]
+
+import torch
+import torch.distributed as dist
+from torch.distributed._tensor import DeviceMesh, DTensor, Replicate
+from torch.distributed.tensor.parallel import PairwiseParallel, parallelize_module
+from torch.distributed.tensor.parallel.ddp import pre_dp_module_transform
+
+from torch.nn.parallel import DistributedDataParallel as DDP
+from torch.testing._internal.common_distributed import skip_if_lt_x_gpu
+
+from torch.testing._internal.common_utils import run_tests
+
+from torch.testing._internal.distributed._tensor.common_dtensor import (
+    DTensorTestBase,
+    MLPModule,
+    with_comms,
+)
+
+# Tensor-Parallel degree
+TP_DEGREE = 2
+LR = 3e-5
+
+
+def init_model(device_type, model_parallel_size=TP_DEGREE):
+    torch.manual_seed(0)
+    model = MLPModule(device_type)
+    torch.manual_seed(0)
+    twod_model = MLPModule(device_type)
+    model = DDP(model)
+
+    # 2-D mesh is [dp, tp]
+    world_size = dist.get_world_size()
+    twod_mesh = DeviceMesh(
+        device_type=device_type,
+        mesh=torch.arange(0, world_size).view(-1, model_parallel_size),
+    )
+
+    dp_pg = twod_mesh.get_dim_groups()[0]
+
+    twod_model = parallelize_module(
+        twod_model, twod_mesh, PairwiseParallel(), tp_mesh_dim=1
+    )
+    pre_dp_module_transform(twod_model)
```

Comment:
I know that DDP/FSDP are two different implementations thus the way to integrate becomes different, but I am wondering in general if we should align these two APIs to be as close as possible?

Reply:
Yes, that's the plan. Once we pass in `DeviceMesh` to both DDP and FSDP, we will merge two.